server/lib/form-prepare.js

```diff
@@ -0,0 +1,46 @@
+import ClientError from './client-error.js';
+
+/**
+ * Creates an array of all the items values from the request object.
+ * @param {object} req, the request.
+ * @param {number} numberOfItems, total number of items to be extracted.
+ * @param {number} recordId, the foreign key to attach all the items to.
+ * @returns an array of items taken from the request body.
+ */
+function createItemsList(req, numberOfItems, recordId) {
+  const items = [];
+  for (let i = 0; i < numberOfItems; i++) {
+    items.push(recordId, req.body[`item${i}`], req.body[`itemCat${i}`], req.body[`itemAmt${i}`]);
+  }
+  if (items.includes(undefined) || items.includes(null)) throw new ClientError(400, 'Incomplete form.');
+  return items;
+}
+
+/**
+ * Creates the SQl for inserting a dynamic amount of rows into the items table.
+ * @param {number} numberOfItems being inserted into the table.
+ * @returns SQL string.
+ */
+function createItemsSql(numberOfItems) {
+  const values = (numberOfItems) => {
+    let stringValues = '(';
+    for (let i = 0; i < (numberOfItems * 4); i++) {
+      if ((i + 1) === Number(numberOfItems * 4)) {
```

Comment:
Why convert to Number here? When you multiply by 4 it already converts it to a number. You probably want to check if `numberOfItems` is a number up at the top, or before calling this function.